gcloud/datastore/test_client.py

```diff
@@ -361,51 +371,26 @@ def test_get_multi_miss_w_deferred(self):
         self.assertEqual([def_key.to_protobuf() for def_key in deferred],
                          [key.to_protobuf()])
 
-    def _verifyProtobufCall(self, called_with, URI, conn):
-        self.assertEqual(called_with['uri'], URI)
-        self.assertEqual(called_with['method'], 'POST')
-        self.assertEqual(called_with['headers']['Content-Type'],
-                         'application/x-protobuf')
-        self.assertEqual(called_with['headers']['User-Agent'],
-                         conn.USER_AGENT)
-
     def test_get_multi_w_deferred_from_backend_but_not_passed(self):
         from gcloud.datastore import _datastore_v1_pb2 as datastore_pb
-        from gcloud.datastore.connection import Connection
+        from gcloud.datastore.entity import Entity
         from gcloud.datastore.key import Key
-        from gcloud.datastore import test_connection
-
-        # Shortening name, import line above was too long.
-        cmp_key_after_req = test_connection._compare_key_pb_after_request
 
         key1 = Key('Kind', dataset_id=self.DATASET_ID)
+        key1_pb = key1.to_protobuf()
         key2 = Key('Kind', 2345, dataset_id=self.DATASET_ID)
-        key_pb1 = key1.to_protobuf()
-        key_pb2 = key2.to_protobuf()
-
-        # Build mock first response.
-        rsp_pb1 = datastore_pb.LookupResponse()
-        entity1 = datastore_pb.Entity()
-        entity1.key.CopyFrom(key_pb1)
-        # Add the entity to the "found" part of the response.
-        rsp_pb1.found.add(entity=entity1)
-        # Add the second key to the "deferred" part of the response.
-        rsp_pb1.deferred.add().CopyFrom(key_pb2)
-
-        # Build mock second response.
-        rsp_pb2 = datastore_pb.LookupResponse()
-        # Add in entity that was deferred.
-        entity2 = datastore_pb.Entity()
-        entity2.key.CopyFrom(key_pb2)
-        rsp_pb2.found.add(entity=entity2)
-
-        connection = Connection()
-        client = self._makeOne(connection=connection)
-        # Add mock http object to connection with response from above.
-        http = connection._http = _HttpMultiple(
-            ({'status': '200'}, rsp_pb1.SerializeToString()),
-            ({'status': '200'}, rsp_pb2.SerializeToString()),
-        )
+        key2_pb = key2.to_protobuf()
+
+        entity1_pb = datastore_pb.Entity()
+        entity1_pb.key.CopyFrom(key1_pb)
+        entity2_pb = datastore_pb.Entity()
+        entity2_pb.key.CopyFrom(key2_pb)
+
+        creds = object()
+        client = self._makeOne(credentials=creds)
+        # mock up two separate requests
+        client.connection._add_lookup_result([entity1_pb], deferred=[key2_pb])
+        client.connection._add_lookup_result([entity2_pb])
```

Comment:
This is the decoupling / simplification:  the old test relies on too much "real" connection behavior, instead of stubbing things out.